client/truncate_log.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Usage: truncate_log.sh <size_threshold_mb> <log_files...>
+# Example: truncate_log.sh 10 /results/log1.log /results/log2.log
+
+set -euo pipefail
+
+if [ "$#" -lt 2 ]; then
+    echo "Usage: $0 <size_threshold_mb> <log_files...>"
+    exit 1
+fi
+
+SIZE_THRESHOLD_MB=$1
+shift
+HEAD_SIZE=$((SIZE_THRESHOLD_MB * 9/10 * 1024 * 1024))
+TAIL_SIZE=$((SIZE_THRESHOLD_MB * 1/10 * 1024 * 1024))
+
+for log_file in "$@"; do
+    if [ ! -f ${log_file} ]; then
+        continue
+    fi
+    
+    LOG_SIZE_MB=$(du -m ${log_file} 2>/dev/null | cut -f1 || echo 0)
```

Comment:
```
$(( $(stat -c %s "${log_file}" 2>/dev/null) / 1024 / 1024 ))
```
is clearer because it doesn't rely on `cut` and `echo`.  It should be as fast as `du -m` because it also uses the file metadata.